File: tecjobs-api/src/main/java/com/raffdevs/tecjobsapi/services/CompanyService.java
package com.raffdevs.tecjobsapi.services;

import com.raffdevs.tecjobsapi.dtos.CreateCompanyDTO;
import com.raffdevs.tecjobsapi.dtos.UpdateCompanyDTO;
import com.raffdevs.tecjobsapi.entities.Company;
import com.raffdevs.tecjobsapi.exceptions.ResourceNotFoundException;
import com.raffdevs.tecjobsapi.models.CompanyModel;
import com.raffdevs.tecjobsapi.models.mapper.CompanyMapper;
import com.raffdevs.tecjobsapi.repositories.CompanyRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.logging.Logger;
import java.util.stream.Collectors;

@Service
public class CompanyService {
    private final Logger logger = Logger.getLogger(CompanyService.class.getName());

    @Autowired
    private CompanyRepository repository;

    public List<CompanyModel> findAll() {
        logger.info("Finding all companies!");
        return repository.findAll()
                .stream()
                .map(CompanyMapper::parseToModel)
                .collect(Collectors.toList());
    }

    public CompanyModel findById(Long id) {
        logger.info("Finding one company!");
        var entity =  repository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("No records for this ID!"));

        return CompanyMapper.parseToModel(entity);
    }

    public CompanyModel create(CreateCompanyDTO data) {
        logger.info("Creating company!");
        Company company = new Company();
        company.setName(data.getName());
        company.setAboutUs(data.getAboutUs());

        return CompanyMapper.parseToModel(repository.save(company));
    }

    public CompanyModel update(UpdateCompanyDTO data) {
        logger.info("Updating company!");

        var company = repository.findById(data.getId())
                .orElseThrow(() -> new ResourceNotFoundException("No records for this ID!"));

        company.setName(data.getName());
        company.setAboutUs(data.getAboutUs());

        return CompanyMapper.parseToModel(repository.save(company));

    }

    public void delete(Long id) {
        logger.info("Deleting one company!");

        var company = repository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("No records for this ID!"));

        repository.delete(company);
    }

}

File: tecjobs-api/TecJobsAPI/Services/CompanyService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using TecJobsAPI.DTO;
using TecJobsAPI.Entities;
using TecJobsAPI.Exceptions;
using TecJobsAPI.Repositories;

namespace TecJobsAPI.Services
{
    public class CompanyService
    {
        private readonly CompanyRepository _repository;

        public CompanyService(CompanyRepository repository)
        {
            _repository = repository;
        }


        public async Task<List<Company>> GetAllCompanies()
        {
            try
            {
                var companies = await _repository.GetAll();
                return companies;
            }
            catch (Exception ex)
            {
                throw new ExceptionReponse("A internal error ocurred!", true);
            }
        }

        public async Task<Company> GetCompanyById(int id)
        {
            try
            {
                var company = await _repository.GetById(id);

                if (company == null)
                {
                    throw new ExceptionReponse("No records found for this ID!", false);
                }
                return company;
            }
            catch (Exception ex)
            {

                throw new ExceptionReponse("A internal error ocurred!", true);
            }
        }

        public async Task<Company> CreateCompany(CreateCompanyDTO data)
        {
            try
            {
                var result = await _repository.Create(data);

                if (result == null)
                {
                    throw new ExceptionReponse("Error on save company!", false);
                }

                return result;
            }
            catch (Exception ex)
            {
                throw new ExceptionReponse("A internal error ocurred!", true);

            }
        }

        public async Task<Company> UpdateCompany(int id, UpdateCompanyDTO data)
        {
            try
            {
                var result = await _repository.Update(id, data);

                if (result == null)
                {
                    throw new ExceptionReponse("No records found for this ID!", false);
                }

                return result;
            }
            catch (Exception ex)
            {
                throw new ExceptionReponse("A internal error ocurred!", true);

            }
        }

        public async Task<Boolean> DeleteCompany(int id)
        {
            try
            {
                var result = await _repository.Delete(id);

                if (!result)
                {
                    throw new ExceptionReponse("No records found for this ID", false);
                }

                return result;
            }
            catch (Exception ex)
            {
                throw new ExceptionReponse("A internal error ocurred!", true);

            }
        }
    }
}